nondex-test/src/test/java/edu/illinois/nondex/core/AbstractCollectionTest.java

```diff
@@ -56,16 +58,22 @@ protected void assertParameterized(T ds, Object derived, String str) {
         }
     }
 
-    protected void assertEqualstUnordered(String msg, String expected, String actual) {
-        Assert.assertEquals(msg + ": " + expected + " =/= " + actual, expected.length(), actual.length());
-        String trimmed = expected.substring(1, expected.length() - 1);
+    private String[] formatString(String msg) {
+        String trimmed = msg.substring(1, msg.length() - 1);
         String[] elems = trimmed.split(",");
-        // TODO(gyori): fix and make this more robust. It does not check duplicates, substrings, etc.
         for (int i = 0; i < elems.length; i++) {
             elems[i] = elems[i].trim();
-            Assert.assertTrue(msg + ": " + trimmed + " =/= " + actual, actual.contains(elems[i]));
         }
+        return elems;
+    }
 
+    protected void assertEqualstUnordered(String msg, String expected, String actual) {
+        String trimmed = expected.substring(1, expected.length() - 1);
+        String[] actualTokenized = this.formatString(actual);
```

Comment:
Is `formatString` the best name? The method seems to be splitting a collection into elements. Even the input is not an arbitrary `msg`.